bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/io/RefreshingOAuth2CredentialsInterceptor.java

```diff
@@ -119,7 +120,13 @@ public RefreshingOAuth2CredentialsInterceptor(ExecutorService scheduler, OAuth2C
 
       @Override
       public void start(Listener<RespT> responseListener, Metadata headers) {
-        HeaderToken token = store.getHeader(TIMEOUT_SECONDS);
+        int timeoutSeconds = TIMEOUT_SECONDS;
+        if (callOptions.getDeadline() != null) {
+          long deadlineMs = callOptions.getDeadline().timeRemaining(TimeUnit.MILLISECONDS);
+          int deadlineSeconds = (int) Math.round(deadlineMs / 1000.0);
```

Comment:
I think the deadlines should be kept in ms

Reply:
The timeout shouldn’t be > deadline. I think would be more appropriate

Reply:
Ack.